### ISSUE TITLE
Use new "expires" option to clean up Tugboat previews

### ISSUE THREAD

**Issue by quicksketch** [backdrop-contrib/tugboat#13]
A brand new (as of yesterday) feature was added to Tugboat to automatically clean up previews a particular time. When a preview is created, you may now specify an `expires` property. Tugboat itself will automatically remove the preview at the expiration time.

Documentation link: https://api.tugboat.qa/v3#tag/Previews/paths/~1previews/post

This seems like a great way for us to handle clean-up of previews, especially as of today we ran into an issue where we had 200+ previews and had expended our total allowed space. I think there may be an issue with the way our preview IDs are stored in config, they can be wiped out when doing config syncs. We should be using `state_set()` or a dedicated table instead, but this new Tugboat feature actually makes keeping track of our previews entirely unnecessary instead, which seems ideal.

**Comment by quicksketch** [comment]
I filed a PR at https://github.com/backdrop-contrib/tugboat/pull/14, but I discovered that Tugboat module tracks existing sites in a helpful report page, with a delete all button for good measure. I'd hate to remove that useful feature.

So my PR is bigger than I thought because instead of just removing the site tracking, I switched it to using the state system instead of config. I still added the new "expires" property, so Tugboat will now clean them up automatically. The cron job however is still needed to keep the Backdrop preview list in sync with what's in Tugboat, so I left it unchanged, other than the state/config switch.
